opentelemetry/proto/profiles/v1experimental/profiles.proto

```diff
@@ -120,42 +129,133 @@ message ResourceProfiles {
+  //   [["allocations","count"], ["space","bytes"]],
+  // If one of the values represents the number of events represented
+  // by the sample, by convention it should be at index 0 and use
+  // sample_type.unit == "count".
+  repeated ValueType sample_type = 1;
+  // The set of samples recorded in this profile.
+  repeated Sample sample = 2;
+  // Mapping from address ranges to the image/binary/library mapped
+  // into that address range.  mapping_table[0] will be the main binary.
+  repeated Mapping mapping_table = 3;
+  // Locations referenced by samples via location_indices.
+  repeated Location location_table = 4;
+  // Functions referenced by locations.
+  repeated Function function_table = 5;
+  // A common table for strings referenced by various messages.
+  // string_table[0] must always be "".
+  repeated string string_table = 6;
+  // frames with Function.function_name fully matching the following
+  // regexp will be dropped from the samples, along with their successors.
+  int32 drop_frames_strindex = 7;   // Index into string table.
+  // frames with Function.function_name fully matching the following
+  // regexp will be kept, even if it matches drop_frames.
+  int32 keep_frames_strindex = 8;  // Index into string table.
+
+  // The following pprof fields are informational, do not affect
+  // interpretation of results.
+
+  // start_time_unix_nano is the start time of the profile.
+  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+  // This field is semantically required.
+  fixed64 start_time_unix_nano = 9;
+  // Duration of the profile, if a duration makes sense.
+  int64 duration_nanos = 10;
+  // The kind of events between sampled occurrences.
+  // e.g [ "cpu","cycles" ] or [ "heap","bytes" ]
+  ValueType period_type = 11;
+  // The number of events between sampled occurrences.
+  int64 period = 12;
+  // Free-form text associated with the profile. The text is displayed as is
+  // to the user by the tools that read profiles (e.g. by pprof). This field
+  // should not be used to store any machine-readable information, it is only
+  // for human-friendly content. The profile must stay functional if this field
+  // is cleaned.
+  repeated int32 comment_strindices = 13; // Indices into string table.
+  // Index into the string table of the type of the preferred sample
+  // value. If unset, clients should default to the last sample value.
+  int32 default_sample_type_strindex = 14;
+
+  reserved 15; // doc_url field is not inherited from pprof.
+
+  // In the event that future pprof versions add additional fields, then
+  // future OpenTelementry versions MAY use these for corresponding purposes.
+  reserved 16,17,18,19;
+
+
+  // second section: OpenTelemetry fields.
+
   // A globally unique identifier for a profile. The ID is a 16-byte array. An ID with
   // all zeroes is considered invalid.
   //
   // This field is required.
-  bytes profile_id = 1;
+  bytes profile_id = 20;
```

Comment:
I have mixed feelings about this field being required. I suspect this is modeled after other OTel signals, but I'm thinking of something like, say, a particular language runtime decides to generate its profiles in the OTel format. Say, Python. Requiring a low-level language runtime to take a dependency on a UUID generator seems excessive and unnecessary. Maybe it's a lesser deal than I imagine.

BTW this also means that pprof profiles cannot be converted into OTel profiles with just the information that is in the pprof profile - this ID will have to be "invented" at the conversion time. It's probably OK, but this is the first field of this kind, so I wanted to call this out.